### PR TITLE
Added typing annotations to transforms/autoaugment

### PR DESCRIPTION
**Pull request by frgfm**
Following up on #2025, this PR adds missing typing annotations in transforms/autoaugment.py.

Any feedback is welcome!

### CODE REVIEW

**Comment by frgfm** [comment]
FYI: I removed the ignore from mypy.ini locally and the checks are all passing on my end now :ok_hand:

File: torchvision/transforms/autoaugment.py
Comment (oke-aditya) [inline torchvision/transforms/autoaugment.py:172]: I think JIT is not happy here. Probably you might need to revert the line.

**Review by oke-aditya — COMMENTED** [review]
:smile: Let's go with the JIT logic as pmeier suggested.

File: torchvision/transforms/autoaugment.py
Comment (fmassa) [inline torchvision/transforms/autoaugment.py:6]: I think the `Any` is not being used anymore?

**Review by fmassa — APPROVED** [review]
LGTM, thanks! I have only one minor comment, otherwise good to merge once CI passes

**Comment by NicolasHug** [comment]
before merging, we should please remove this file from mypy's ignore list so that the CI can check it as mentioned in https://github.com/pytorch/vision/issues/2025#issuecomment-889989524

**Comment by frgfm** [comment]
> before merging, we should please remove this file from mypy's ignore list so that the CI can check it as mentioned in [#2025 (comment)](https://github.com/pytorch/vision/issues/2025#issuecomment-889989524)

Correct, but I thought we could merge https://github.com/pytorch/vision/pull/4236 before (otherwise we're gonna get conflicts about the mypy.ini on all related PRs), then I'll merge master and edit on this branch (+ the edit suggested by @fmassa )

**Comment by NicolasHug** [comment]
That sounds good to me, as long as we don't forget the CI :)

File: torchvision/transforms/autoaugment.py
Comment (datumbox) [reply torchvision/transforms/autoaugment.py:6]: Francisco is right:
```suggestion
from typing import List, Tuple, Optional, Dict
```

**Pull request merged**
